### ISSUE TITLE
don’t prompt to add our `.zshrc` one liner if it’s already there

### ISSUE THREAD

**Issue by mxcl** [teaxyz/setup#52]
bit tricky

**Comment by jhheider** [comment]
best way is to just add `source $TEA_PREFIX/zshrc` (or similar), and then grep for it to see if it's missing.

**Comment by mxcl** [comment]
> best way is to just add source $TEA_PREFIX/zshrc (or similar), and then grep for it to see if it's missing.

Yeah I guess the installer should create `$TEA_PREFIX/tea.xyz/etc/hook.zsh` and source that, then it's a consistent line that we add and can grep for (with tolerant whitespace checks).

> If we eventually add the hook command to teaxyz/cmd, then source $(tea hook zsh) is even more future-proof.

Probably will happen

**Comment by jedahan** [comment]
Just a heads up, if you go down the source route, please make sure the invocation is as fast as sourcing a static file.
Its very easy for it to slow down shell startup.

```shell
 ▲ tea hyperfine --shell=none "tea --help" "zsh -c exit"
Benchmark 1: tea --help
  Time (mean ± σ):      50.3 ms ±   0.7 ms    [User: 43.8 ms, System: 5.5 ms]
  Range (min … max):    49.3 ms …  51.8 ms    59 runs
 
Benchmark 2: zsh -c exit
  Time (mean ± σ):       3.4 ms ±   0.3 ms    [User: 1.2 ms, System: 0.8 ms]
  Range (min … max):     3.0 ms …   6.7 ms    631 runs
  
Summary
  'zsh -c exit' ran
   14.76 ± 1.36 times faster than 'tea --help'
```

**Comment by mxcl** [comment]
Being as fast as possible is certainly a goal, though at this v0 stage we're forgiving of slow down. For me in use we aren't yet noticeably slow.

Not really sure specifically what you are talking about since this ticket is unrelated, you mean the one-liner should source a static file as much as possible? Certainly we could cache the output but it must be computed in general.

**Comment by mxcl** [comment]
Fixed in 0.11.0
